gitlab/resource_gitlab_project_approval_rule.go

```diff
@@ -0,0 +1,429 @@
+		_, err = client.Projects.DeleteProjectApprovalRule(project, ruleIDInt)
+	}
+
+	return err
+}
+
+// getApprovalRuleByID checks the list of rules and finds the one that matches our rule ID.
+func getApprovalRuleByID(meta interface{}, id string) (*gitlab.ProjectApprovalRule, error) {
+	var rules []*gitlab.ProjectApprovalRule
+	var err error
+
+	projectID, ruleIDStr, err := splitApprovalRuleID(id)
+	if err != nil {
+		return nil, err
+	}
+
+	client := meta.(*gitlab.Client)
+	if rules, _, err = client.Projects.GetProjectApprovalRules(projectID); err == nil {
+		var ruleIDInt int
+
+		if ruleIDInt, err = strconv.Atoi(ruleIDStr); err != nil {
+			return nil, err
+		}
+
+		for _, r := range rules {
+			if r.ID == ruleIDInt {
+				log.Printf("[DEBUG] found project-level rule %+v", r)
+				return r, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+// createApprovalRuleID creates an ID of two parts: projectID:ruleID
+func createApprovalRuleID(ruleID int, projectID string) string {
+	return buildTwoPartID(gitlab.String(projectID), gitlab.String(strconv.Itoa(ruleID)))
+}
+
+// splitApprovalRuleID splits an approval rule into two parts, projectID:ruleID.
+// An error is returned if one occurs.
+func splitApprovalRuleID(ruleID string) (string, string, error) {
+	return parseTwoPartID(ruleID)
+}
+
+// flattenApprovalRuleUserIDs flattens a list of approval user ids into a list
+// of ints for storage in state.
+func flattenApprovalRuleUserIDs(users []*gitlab.BasicUser) []interface{} {
+	m := []interface{}{}
+	for _, user := range users {
+		m = append(m, user.ID)
+	}
+
+	return m
+}
+
+// flattenApprovalRuleGroupIDs flattens a list of approval group ids into a list
+// of ints for storage in state.
+func flattenApprovalRuleGroupIDs(groups []*gitlab.Group) []interface{} {
+	m := []interface{}{}
+	for _, group := range groups {
+		m = append(m, group.ID)
+	}
+
+	return m
+}
+
+// expandApproverIds Expands an interface into a list of ints to read from state.
+func expandApproverIds(ids interface{}, hasItems bool) []int {
+	m := []int{}
+	if hasItems {
+		for _, id := range ids.(*schema.Set).List() {
+			m = append(m, id.(int))
+		}
+	}
+
+	return m
+}
+
+// getProjectApprovalRuleUserIDs generates a list of approvers for a rule that
+// could not be found in the project or in any attached groups and returns a
+// list containing only the user IDs not found as developer <= higher in the
+// project, or developer <= in any of the rule groups.
+//
+// This method attempts to work around a bug in GitLab where there are instances
+// where the `users` attribute in the response will be empty under certain
+// conditions.
+//
+// Reference: https://gitlab.com/gitlab-org/gitlab/issues/35008
+func getProjectApprovalRuleUserIDs(client *gitlab.Client, ruleID string) ([]int, error) {
+	log.Print("[DEBUG] Looking up project approval rule User IDs as a workaround for bug in GitLab.")
+
+	userIDs := newMemberList([]int{})
+
+	rule, err := getApprovalRuleByID(client, ruleID)
```

Comment:
You shouldn't need to make this API call here, since you have already made this API call previously in the `resourceGitlabProjectApprovalRuleRead` function.